packages/react-google-maps-api/src/components/addons/InfoBox.tsx

```diff
@@ -0,0 +1,186 @@
+  zIndex?: number;
+  /** This event is fired when the close button was clicked. */
+  onCloseClick?: () => void;
+  /** This event is fired when the <div> containing the InfoBox's content is attached to the DOM. You may wish to monitor this event if you are building out your info window content dynamically. */
+  onDomReady?: () => void;
+  /** This event is fired when the content property changes. */
+  onContentChanged?: () => void;
+  /** This event is fired when the position property changes. */
+  onPositionChanged?: () => void;
+  /** This event is fired when the InfoBox's zIndex changes. */
+  onZindexChanged?: () => void;
+  /** This callback is called when the infoBox instance has loaded. It is called with the infoBox instance. */
+  onLoad?: (infoBox: GoogleMapsInfoBox) => void;
+  /** This callback is called when the component unmounts. It is called with the infoBox instance. */
+  onUnmount?: (infoBox: GoogleMapsInfoBox) => void;
+}
+
+export class InfoBoxComponent extends React.PureComponent<InfoBoxProps, InfoBoxState> {
+  static contextType = MapContext;
+
+  registeredEvents: google.maps.MapsEventListener[] = []
+  containerElement: HTMLElement | null = null
+
+  state: InfoBoxState = {
+    infoBox: null,
+  };
+
+  open = (infoBox: GoogleMapsInfoBox, anchor?: google.maps.MVCObject) => {
+    if (anchor) {
+      infoBox.open(this.context, anchor)
+    } else if (infoBox.getPosition()) {
+      // @ts-ignore
+      infoBox.open(this.context)
+    } else {
+      invariant(
+        false,
+        "You must provide either an anchor or a position prop for <InfoBox>."
+      )
+    }
+  };
+
+  setInfoBoxCallback = () => {
+    const { anchor, onLoad } = this.props
+    const { infoBox } = this.state
+
+    if (infoBox !== null && this.containerElement !== null) {
+      infoBox.setContent(this.containerElement)
+      this.open(infoBox, anchor)
+
+      if (onLoad) {
+        onLoad(infoBox)
+      }
+    }
+  };
+
+  componentDidMount() {
+    const { options } = this.props
+
+    let position = options ? options.position : null
+    if (position && !(position instanceof google.maps.LatLng)) {
+      position = new google.maps.LatLng(position.lat, position.lng)
+    }
+
+    // google-maps-infobox uses `google` as a global variable. Since we don't
+    // have `google` on the server, we can not use it in server-side rendering.
+    // As a result, we import google-maps-infobox here to prevent an error on
+    // a isomorphic server
+    delete require.cache[require.resolve('google-maps-infobox')]
+    const InfoBox: typeof GoogleMapsInfoBox = require('google-maps-infobox').default
+
+    // @ts-ignore
```

Comment:
please do not use //@ts-ignore if possible

Reply:
I had similar problem with original marker-clusterer, so I had created new `@react-google-maps/marker-clusterer` copied the code from original, and added types. after that I have added it as dependency into main `@react-google-maps/api`
Can you please copy our marker-clusterer package, and add the code from `google-maps-infobox`, so I will help you to add Typescript to the code, publish it to the npm and so on

Reply:
Sure, I'll try to work on that today

Reply:
@inv8der how is your progress?

Reply:
Good. I just need to fix a few more TypeScript errors, then I'll push my changes. I'll do that tomorrow morning. Should I create a new pull request or just use this one?

Reply:
use this one. please add yourself to contributors list in package.json